services/audio/src/main/java/ca/bc/gov/tno/services/audio/kafka/KafkaAudioProducer.java

```diff
@@ -0,0 +1,187 @@
+import ca.bc.gov.tno.services.audio.models.PublishedRecord;
+
+import ca.bc.gov.tno.ContentType;
+
+/**
+ * KafkaAudioProducer class, provides a process that pushes messages to
+ * Kafka topics.
+ */
+@Async
+@Component
+public class KafkaAudioProducer implements ApplicationListener<ProducerSendEvent> {
+  private static final Logger logger = LogManager.getLogger(KafkaAudioProducer.class);
+
+  private final ApplicationEventPublisher eventPublisher;
+  private final KafkaProducerConfig kafkaConfig;
+  private final IContentReferenceService contentReferenceService;
+  private final KafkaProducer<String, SourceContent> producer;
+
+  /**
+   * Create a new instance of a KafkaAudioProducer object.
+   */
+  @Autowired
+  public KafkaAudioProducer(final KafkaProducerConfig config, final IContentReferenceService service,
+      final ApplicationEventPublisher eventPublisher) {
+    this.kafkaConfig = config;
+    this.contentReferenceService = service;
+    this.eventPublisher = eventPublisher;
+
+    var props = new Properties();
+    props.put(ProducerConfig.CLIENT_ID_CONFIG, kafkaConfig.getClientId());
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaConfig.getBootstrapServers());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+    props.put(ProducerConfig.ACKS_CONFIG, "all");
+    props.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);
+    props.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, 5);
+    props.put(ProducerConfig.RETRIES_CONFIG, 10000000);
+    props.put(ProducerConfig.BATCH_SIZE_CONFIG, 16384);
+    props.put(ProducerConfig.LINGER_MS_CONFIG, 1);
+    props.put(ProducerConfig.BUFFER_MEMORY_CONFIG, 33554432);
+    this.producer = new KafkaProducer<String, SourceContent>(props);
+  }
+
+  /**
+   * Send the message to Kafka.
+   */
+  @Override
+  public void onApplicationEvent(ProducerSendEvent event) {
+    try {
+      var config = event.getConfig();
+      send(event);
+
+      var doneEvent = new TransactionCompleteEvent<>(event.getSource(), config);
+      eventPublisher.publishEvent(doneEvent);
+    } catch (InterruptedException | ExecutionException ex) {
+      var errorEvent = new ErrorEvent(this, ex);
+      eventPublisher.publishEvent(errorEvent);
+      producer.flush();
+    }
+  }
+
+  /**
+   * Push the content stub into Kafka.
+   * 
+   * @param event
+   * @throws ExecutionException
+   * @throws InterruptedException
+   */
+  public void send(ProducerSendEvent event) throws InterruptedException, ExecutionException {
+    var clip = event.getData();
+
+    var config = event.getConfig();
+    var source = config.getId();
+    var topic = config.getTopic();
+
+    try {
+      logger.info(String.format("New audio content extracted: '%s', topic: %s", clip, topic));
+      var contentReference = new ContentReference(source, clip, topic);
```

Comment:
This ContentReference needs to somehow point to the Clip that was created.  It probably should be just the clip name.  That way the Transcription Service can grab the clip once we build it.

Reply:
At the moment I have no uid to put in the uid column of ContentReference, so I'm storing the clip-path as a unique value in the uid field. I can shorten this to the clip name or discuss what you think would be a better solution.